src/stream/stream/all.rs

```diff
@@ -14,6 +14,17 @@ pub struct AllFuture<'a, S, F, T> {
     pub(crate) _marker: PhantomData<T>,
 }
 
+impl<'a, S, F, T> AllFuture<'a, S, F, T> {
+    pub fn new(stream: &'a mut S, f: F) -> Self {
```

Comment:
```suggestion
    pub(crate) fn new(stream: &'a mut S, f: F) -> Self {
```

I don't think this is public at this time, but [public methods can avoid the dead_code warning](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=e613e0113ead88b7038308455ff9523a), so I think it is preferable to make it private (avoid to compile unused private API).